app/src/main/java/com/example/peterchu/watplanner/Models/Schedule/CourseComponent.java

```diff
@@ -195,4 +207,79 @@ public void setInstructors(String[] instructors) {
     public String toString() {
         return String.format("%s %s - %s %s - %s %s~%s", subject, catalogNumber, type, section, day, startTime, endTime);
     }
+
+    public List<WeekViewEvent> toWeekViewEvents(int month) {
+        List<WeekViewEvent> weekViewEvents = new ArrayList<WeekViewEvent>();
+        try {
+            Date eventStartTime = componentDateFormat.parse(this.startTime);
+            Date eventEndTime = componentDateFormat.parse(this.endTime);
+            String eventName = String.format("%s %s %s", subject, catalogNumber, type);
+
+            Calendar date = Calendar.getInstance();
+            date.set(Calendar.MONTH, month);
+            date.set(Calendar.YEAR, 2017);
+            date.set(Calendar.DAY_OF_MONTH, 1);
+
+            Integer dayOfWeek = this.getDayOfWeek();
+            if (dayOfWeek == null || (month < Calendar.MAY) || (month > Calendar.AUGUST)) {
```

Comment:
Can you leave a comment about this? I'm a bit confused, but other than that this is good